internal/datatransfer/monitor.go

```diff
@@ -17,17 +20,24 @@ type Monitor struct {
 	ticker                      *time.Ticker
 	lastSuccessfulSyncTimes     map[string]time.Time
 	lastSuccessfulSyncTimesLock sync.RWMutex
+	fsSync                      FileSync
 }
 
 func NewMonitor(workloadsManager *workload.WorkloadManager, configManager *configuration.Manager) *Monitor {
 	ticker := time.NewTicker(configManager.GetDataTransferInterval())
+
 	monitor := Monitor{
 		workloads:                   workloadsManager,
 		config:                      configManager,
 		ticker:                      ticker,
 		lastSuccessfulSyncTimes:     make(map[string]time.Time),
 		lastSuccessfulSyncTimesLock: sync.RWMutex{},
 	}
+	err := monitor.InitStorage()
```

Comment:
Having the storage initialized once will cause problems when the s3 configuration changes; the Monitor will continue sending files to the old bucket with old credentials while the configuration might specify different ones.
That was the reason, why the `s3` client was created in `syncPaths` function for each synchronization cycle.

The only way to refresh `fsSync` is to restart the device worker, which is not desired.